torch/csrc/dynamo/eval_frame.c

```diff
@@ -299,6 +302,17 @@ typedef struct cache_entry {
   struct cache_entry* next;
 } CacheEntry;
 
+// An abstraction to encapsulate all the information to be saved on the extra
+// scratch space of the code object. This enables us to reuse the same co_extra
+// field, saving extra allocations.
```

Comment:
Well, you're not actually saving any allocations at the moment, since you needed to malloc ExtraState so you net out two allocations (one for the extra scratch, one for ExtraState) whereas before you had two for extra scratch.

Reply:
I will modify the comment. The goal is to not have two allocations for extra scratch. @jansel mentioned that its very expensive to have more allocations for extra. I don't understand why yet. Will try to dig deeper to understand this point.